### ISSUE TITLE
Use of `\MakeUppercase` sometimes causes table of contents compilation to fail

### ISSUE THREAD

**Issue by tymorrow** [kohoman/mstTheses#1]
The use of `\MakeUppercase` was causing the use of `\section` to break on my computer for some unknown reason.

My problem was similar to [this][1] and using the solution on that page solved the issue.
Despite being fixed, the nature of the problem remains a mystery to me.

One question I have is "are any arguments for/against using the `\MakeUppercase` command over `\uppercase`?"

[1]: http://tex.stackexchange.com/questions/25694/error-when-trying-to-use-maketextuppercase-to-customize-the-table-of-contents

**Comment by AP6YC** [comment]
Seven years later, and here we are! I am having the same issue too.

In case this affects anyone else in the future, I ended up having the same problem in a specific scenario with the publication option format: if I have a `\ThesisBodyChapter` in a `ThesisOnlyChapters` section _after_ the `ThesisPublications` environment, then an error arises from using `\MakeUppercase` in the section headings and their TOC entries.

My solution was to directly change the `mstogs.cls` definition of `\ThesisBodyChapter` to use `\uppercase` instead of `\MakeUppercase`. ([This file](https://github.com/kohoman/mstTheses/blob/fd185bf028c13f1a93482e42a825335656230f6c/template_pto/mstogs.cls#L488), changing [here](https://github.com/kohoman/mstTheses/blob/fd185bf028c13f1a93482e42a825335656230f6c/template_pto/mstogs.cls#L492) and [here](https://github.com/kohoman/mstTheses/blob/fd185bf028c13f1a93482e42a825335656230f6c/template_pto/mstogs.cls#L500) to use `\uppercase` instead of `\MakeUppercase`).

I don't know about suggesting a PR for this because even I don't have a satisfying answer for why packages should use one over the other, especially if the "safer" `\MakeUppercase` is known to break in situations like this. Apparently `\uppercase` is the TeX primitive that is defined for pretty much the ASCII set of Latin characters, and LaTeX macros like `\MakeUppercase` are meant fill the gap for Unicode and other character sets. I think this is a case where the macro's behavior, expansion timing, etc. in different LaTeX distributions and versions changes and breaks things from section hrefs down to lowly thesis classes like this one.

Long story short, I only use Latin characters in my headings, so patching in `\uppercase` gets the document to compile again.
